libcudacxx/test/libcudacxx/std/numerics/c.math/fp_traits.pass.cpp

```diff
@@ -0,0 +1,458 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+// <cmath>
+
+#include <cuda/std/cassert>
+#include <cuda/std/cmath>
+#include <cuda/std/limits>
+#include <cuda/std/type_traits>
+
+#include "fp_compare.h"
+#include "test_macros.h"
+
+template <class T>
+__host__ __device__ void test_fpclassify(float val)
+{
+  static_assert((cuda::std::is_same<decltype(cuda::std::fpclassify(val)), int>::value), "");
+  assert(cuda::std::fpclassify(T(val)) == FP_NORMAL);
+  assert(cuda::std::fpclassify(T(1.0)) == FP_NORMAL);
+  assert(cuda::std::fpclassify(T(0.0)) == FP_ZERO);
+  assert(cuda::std::fpclassify(T(-1.0)) == FP_NORMAL);
+  assert(cuda::std::fpclassify(T(0.0)) == FP_ZERO);
```

Comment:
Suggestion: This test appears twice, maybe this was intended:
```suggestion
  assert(cuda::std::fpclassify(T(-0.0)) == FP_ZERO);
```